강산아/boj/week13/Gold_IV_1753.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Author    : Kang San Ah
+ * Date      : 2024.08.13(Wed)
+ * Runtime   : 1 sec
+ * Memory    : 256 MB
+ * Algorithm : Greedy Algorithm, Dijkstra
+ */
+
+class Edge implements Comparable<Edge>{
+    public int vex;
+    public int cost;
+
+    public Edge(int vex, int cost){
+        this.vex = vex;
+        this.cost = cost;
+    }
+
+    @Override
+    public int compareTo(Edge o) {
+        return this.cost - o.cost;
+    }
+}
+
+public class Main {
+    private static int[] dis;
+    static ArrayList<ArrayList<Edge>> graph;
+    static int V,E,K;
+
+    public void solution(int v){
+        PriorityQueue<Edge> pq = new PriorityQueue<>();
+        pq.offer(new Edge(v, 0));
+        dis[v] = 0;
+
+        while (!pq.isEmpty()){
+            Edge tmp = pq.poll();
+            int now = tmp.vex;
+            int nowCost = tmp.cost;
+            if (nowCost > dis[now]) continue;
+            for (Edge ob : graph.get(now)){
+                if (dis[ob.vex] > nowCost+ob.cost){
+                    dis[ob.vex] = nowCost + ob.cost;
+                    pq.offer(new Edge(ob.vex , nowCost + ob.cost));
+                }
+            }
+        }
+    }
+
+    public static void main(String[] args){
+
+        Main main= new Main();
+
+        graph = new ArrayList<>();
+
+        Scanner scanner = new Scanner(System.in);
+        V = scanner.nextInt();
+        E = scanner.nextInt();
+        K = scanner.nextInt();
+
+        for (int i = 0 ; i <= V ; i++){
+            graph.add(new ArrayList<Edge>());
+        }
+
+        dis = new int[V+1];
+        Arrays.fill(dis,Integer.MAX_VALUE);
```

Comment:
```suggestion
        dis = new int[V+1];
        
        for (int i = 0 ; i <= V ; i++){
            graph.add(new ArrayList<Edge>());
            dis[i] = Integer.MAX_VALUE;
        }
```

어차피 그래프 요소 인스턴스 생성하는 반복문이 돌아야 하니까 fill() 대신에 같은 반복문 내에서 처리하면 더 좋을거 같습니다.